gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -182,6 +185,8 @@ def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     # This wrapper is very slow, so it will subsequently re-wrap with a much faster wrapper
     # (or unwrap back to the original method).
     new_method = nil
+    # this prevents us from running the final checks twice for every method def.
+    T::Private::DeclState.current.skip_next_on_method_added = true
```

Comment:
This seems like a footgun. Do you think we could store a bit on the sig / current_declaration that says whether we did the final checks for a method already? I'm worried that skipping the _entire_ `_on_method_added` hook is going to be something we regret in the future if the only thing we're trying to prevent is the final method checks that happen in `_on_method_added`.

Reply:
It seems like the `_on_method_added` hook is called three times for every method definition: first when you say `def`, second, when we wrap the method in the lazy sig checking logic (this diff), and third, when we unwrap the lazy sig checking logic and wrap it with faster logic ([here](https://github.com/sorbet/sorbet/pull/1181/files#diff-7f2b312a014dc515dc529a2b66e8a714)).

Those last two calls were "not noticed", or should I say, were essentially no-ops, because what used to be the very first thing we do in `_on_method_added` was check to see if there was a `current.active_declaration`, and return if there wasn't one. Now, because we want the final checks to be run for every method def, not just ones with sigs, I do the final checks before looking to see if there was or wasn't a `current.active_declaration`.

The way I've been communicating that I don't want the final checks to be run again is with this global flag that says whether to skip the next `_on_method_added`. I'm pretty sure that doing it this way doesn't cause us to start skipping doing things that we were not already skipping. But, it is a bit of a coarse way of saying we want to skip just the final checks. If we want to put a bit somewhere, I still think it's have to deal with `DeclState`.

Reply:
I'm not concerned about skipping things that we were already doing, but preventing a footgun interfering with a future refactor. i.e., someone expecting that they could put something in `_on_method_added` but then not seeing it run because some boolean was set somewhere else.